### PR TITLE
[Valet 4.0] Set up code coverage

### LINKED ISSUE #201: Integrate Codecov tracking
Enabling code coverage tracking via codecov.io will help us with finding dead code, and also help enforce good testing practices.

The integration has already been enabled on codecov. This issue tracks integrating coverage data into our CI. Our upload token is 5165deef-da9c-443d-90ea-bb0620bffe44.

### PR DESCRIPTION
**Pull request by dfed**
Resolves #201. I decided to make this change against Valet 4.0 in order to prevent additional merge conflicts.

### CODE REVIEW

**Comment by dfed** [comment]
Looks like we’ve got some low-hanging fruit to increase test coverage, though we’ll always have trouble upping our Mac-specific coverage due to code-signing restrictions.

Good news is: this PR is ready for review.

File: Scripts/build.swift [hunk 1]
```diff
@@ -131,6 +131,35 @@ enum Platform: String, CustomStringConvertible {
...
+        switch self {
+        case .tvOS_10,
+             .tvOS_11,
+             .tvOS_12,
+             .tvOS_13,
+             .macOS_10_12,
+             .macOS_10_13,
+             .macOS_10_14,
+             .macOS_10_15:
+            // We only enable code coverage on our tvOS and macOS targets.
```
Comment (NickEntin) [inline Scripts/build.swift]: Why only tvOS and macOS?  Can we do iOS as well?

Comment (dfed) [reply Scripts/build.swift]: When I started writing this PR, I didn't have a way to run coverage on multiple test targets. Now we do, so enabling tests on all platforms that support it seems reasonable. I'll update!

Comment (dfed) [reply Scripts/build.swift]: Fixed in https://github.com/square/Valet/pull/202/commits/9e580e544c9c66671692b2e674f6105489315a3a.

**Pull request merged**